src/hoverifier.test.ts

```diff
@@ -330,6 +339,68 @@ describe('Hoverifier', () => {
         }
     })
 
+    it('highlights document highlights', async () => {
+        for (const codeViewProps of testcases) {
+            const hoverifier = createHoverifier({
+                closeButtonClicks: NEVER,
+                hoverOverlayElements: of(null),
+                hoverOverlayRerenders: EMPTY,
+                getHover: createStubHoverProvider(),
+                getDocumentHighlights: createStubDocumentHighlightProvider([
+                    { range: { start: { line: 24, character: 9 }, end: { line: 4, character: 15 } } },
+                    { range: { start: { line: 45, character: 5 }, end: { line: 45, character: 11 } } },
+                    { range: { start: { line: 120, character: 9 }, end: { line: 120, character: 15 } } },
+                ]),
+                getActions: () => of(null),
+                pinningEnabled: true,
+                documentHighlightClassName: 'test-highlight',
+            })
+            const positionJumps = new Subject<PositionJump>()
+            const positionEvents = of(codeViewProps.codeView).pipe(
+                findPositionsFromEvents({ domFunctions: codeViewProps })
+            )
+
+            const codeViewSubscription = hoverifier.hoverify({
+                dom: codeViewProps,
+                positionEvents: NEVER,
+                positionJumps: NEVER,
+                resolveContext: () => {
+                    throw new Error('not called')
+                },
+            })
+            hoverifier.hoverify({
+                dom: codeViewProps,
+                positionEvents,
+                positionJumps,
+                resolveContext: () => codeViewProps.revSpec,
+            })
+
+            dispatchMouseEventAtPositionImpure('mouseover', codeViewProps, {
+                line: 24,
+                character: 6,
+            })
+
+            await hoverifier.hoverStateUpdates
+                .pipe(
+                    filter(state => !!state.hoverOverlayProps),
+                    first()
+                )
+                .toPromise()
+
+            codeViewSubscription.unsubscribe()
+
+            assert.isDefined(hoverifier.hoverState.hoverOverlayProps)
+            await of(null).pipe(delay(200)).toPromise()
```

Comment:
Can the test work without a timeout? Shouldn't all the highlighting be done by the time the hover overlay is shown (i.e. props are not `undefined`)?

Reply:
They work on different subscriptions, so the timeout is necessary to make sure that the subscriber gets called. Removing makes the test run too fast.

Reply:
Would probably be solveable somehow with the TestScheduler, but I don't have good advice here. The tests are not the cleanest in general in this file

Reply:
![image](https://user-images.githubusercontent.com/10532611/86403960-fe008600-bcae-11ea-9c7b-d3f90d22cd02.png)